libs/isograph-react/src/tests/__isograph/Query/subquery/resolver_reader.ts

```diff
@@ -0,0 +1,47 @@
+import type { EagerReaderArtifact, ReaderAst } from '@isograph/react';
+import { Query__subquery__param } from './param_type';
+import { Query__subquery__output_type } from './output_type';
+import { subquery as resolver } from '../../../normalizeData.test';
+
+const readerAst: ReaderAst<Query__subquery__param> = [
+  {
+    kind: "Linked",
+    fieldName: "query",
+    alias: null,
+    arguments: null,
+    concreteType: "Query",
+    selections: [
+      {
+        kind: "Linked",
+        fieldName: "node",
+        alias: null,
+        arguments: [
+          [
+            "id",
+            { kind: "Variable", name: "id" },
+          ],
+        ],
+        concreteType: null,
```

Comment:
I don't think we need the `__typename` in the reader AST. Instead, we should make the type of the linked field be `NonNullLink`, e.g. `{ Query: { node____id___3: { __link: '3', __typename: 'User' } }` or whatnot

Reply:
Yes, this is something I've asked about before, and you wanted to generate it in `readerAst`, it's all about some trade offs

Storing `__typename` in store is much simpler for us, it removes a bunch of code,
but it leads to storing more data, if we have an array of concrete types, we store `__typename` N times, instead of once.

Reply:
Oh I'm sorry! I was clearly wrong when I said that. Let's store the typename in the store. Sorry about telling you the wrong thing.

If we want, we can later optimize the concrete type situation by also storing a `{ kind: 'AllOneType, typename, records }` struct.

Reply:
If I remove `concreteType` from `readerAst`, then `missingFieldHandler` has to return `__typename`
But we can't calculate `__typename` from `id` to implement `missingFieldHandler` for `Query.node` 😖

Reply:
We need to be able yo store just the `__typename` when normalizing

Something like `Link  | MissingLink` 

```tsx
interface MissingLink {
  __typename: TypeName
}
```

Reply:
Ah... How would we ever support the Node missing field handler, in that case, if the ID is not globally unique?

You could for example write `node(id: $id) { ... on Actor { ... } ... on User { ... } }` :/

Incidentally, the use of the node field is being discouraged, because it's costly on the backend. Since you don't know the typename of the ID, you potentially have to look in many databases.

I wonder if we can specially handle this case, e.g. if your query is of the form `interface_field { ... on ConcreteType { ... } }`, we can provide that concrete type as a param to the missing field handler.

In any case, for now, let's just assume that the ID is a Pet and solve this properly later.

Reply:
> Ah... How would we ever support the Node missing field handler, in that case, if the ID is not globally unique?

The `node` field doesn't make sense when `ID` is not globally unique.
With this PR it's hard to support `node` field when `ID` is globally unique.

Having a single layer cache, and `getRecordId` would allow to work great with `unique` and `not unique` `ID`, I'm curios what are the reasons for us to use two layer cache?